results/docs/Memory_benchmarks.md

```diff
@@ -0,0 +1,34 @@
+# Memory Benchmarks
+
+## Benchmarks Used
+
+The suite of memory benchmarks which is a part of the [university of wisconsin micro-benchmarks suite](https://github.com/VerticalResearchGroup/microbench) and targets specially the L1 cache miss performance can be used to analyse the performance difference between gem5 and real hardware system.
+
+- MC:  Benchmark with load instructions, generates many Conflict Misses.
+- MCS: Benchmark with store instructions, generates many Conflict Misses.
+- MD : Benchmark containing cache resident linked list transversal.
+- MI : Independent memory access stored in static array, this benchmark has good spatial and temporal locality.
+- MIM :Independent memory access stored in static array, this benchmark has very bad spatial locality.
+- MIM2 :Same as MIM, but it has 2 extra coalescing ops in between.
+
+## gem5 results:
+
+<img src="../images/memory_benchmarks_gem5.png" width="300" height="300">
+
+The gem5 results look analytically correct, the MI benchmark which is supposed to have the least missrate theoretically does so as seen in the graph. The MIM and MIM2 benchmarks have bad spatial and temporal locality hence have a very high missrate. The MCS and MC benchmark refelct the cache model beahviour when a conflict miss occurs. The MCS benchmark has more missrate than MC because of the differenc ein store bandwidth, store queue entry size and various other factors in our skylake architecture configuration in gem5.
```

Comment:
```suggestion
The gem5 results look analytically correct. The MI benchmark theoretically has the smallest miss rate, which is shown in the graph. The MIM and MIM2 benchmarks have poor spatial and temporal locality. Hence, these microbenchmarks have a high miss rate. The MCS and MC benchmark reflect the cache model beahvior when a conflict miss occurs. The MCS benchmark has a higher miss rate than MC because of the difference in store bandwidth, store queue entry size, and various other factors in our Skylake architecture configuration in gem5.
```